core/pkg/storage/clusterstorage.go

```diff
@@ -0,0 +1,290 @@
+package storage
+
+import (
+	"bytes"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/opencost/opencost/core/pkg/log"
+	"github.com/opencost/opencost/core/pkg/util/json"
+	"gopkg.in/yaml.v2"
+)
+
+var defaultClusterConfig = ClusterConfig{}
+
+// ClusterStorage is a Storage implementation which connects to a remote file storage over http
+type ClusterStorage struct {
+	client *http.Client
+	host   string
+	port   int
+}
+
+type ClusterConfig struct {
+	Host string `yaml:"host"`
+	Port int    `yaml:"port"`
+}
+
+// parseConfig unmarshals a buffer into a Config with default HTTPConfig values.
+func parseClusterConfig(conf []byte) (ClusterConfig, error) {
+	config := defaultClusterConfig
+	if err := yaml.Unmarshal(conf, &config); err != nil {
+		return ClusterConfig{}, err
+	}
+
+	return config, nil
+}
+
+func NewClusterStorage(conf []byte) (*ClusterStorage, error) {
+	config, err := parseClusterConfig(conf)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewClusterStorageWith(config)
+}
+
+// NewBucketWithConfig returns a new Bucket using the provided s3 config values.
+func NewClusterStorageWith(config ClusterConfig) (*ClusterStorage, error) {
+	host := "localhost"
+	if config.Host != "" {
+		host = config.Host
+	}
+	port := 9006
+	if config.Port != 0 {
+		port = config.Port
+	}
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
```

Comment:
We'll likely want to add configuration for TLS certs, and probably a way to auth?